### ISSUE TITLE
Package repos: Prefer version over SHA in Go to Definition links

### ISSUE THREAD

**Issue by varungandhi-src** [sourcegraph/sourcegraph#31938]
Sub-task of https://github.com/sourcegraph/sourcegraph/issues/31937, see that issue for more context.

**Comment by varungandhi-src** [comment]
@malomarrec I don't think this falls under growth and integrations -- was that label added by mistake?

**Comment by malomarrec** [comment]
Apologies, I was on feedback rotation and had to triage issues. This one
looked like an integration one at first😵‍💫. Do you happen to know what
team it fits in?

**Comment by varungandhi-src** [comment]
Added a `team/code-intelligence` label. 😄
